backend/src/apiserver/storage/pipeline_store.go

```diff
@@ -82,16 +84,26 @@ type PipelineStore struct {
 
 // Runs two SQL queries in a transaction to return a list of matching pipelines, as well as their
 // total_size. The total_size does not reflect the page size.
-func (s *PipelineStore) ListPipelines(opts *list.Options) ([]*model.Pipeline, int, string, error) {
+func (s *PipelineStore) ListPipelines(filterContext *common.FilterContext, opts *list.Options) ([]*model.Pipeline, int, string, error) {
 	errorF := func(err error) ([]*model.Pipeline, int, string, error) {
 		return nil, 0, "", util.NewInternalServerError(err, "Failed to list pipelines: %v", err)
 	}
 
 	buildQuery := func(sqlBuilder sq.SelectBuilder) sq.SelectBuilder {
-		return opts.AddFilterToSelect(sqlBuilder).
-			From("pipelines").
-			LeftJoin("pipeline_versions ON pipelines.DefaultVersionId = pipeline_versions.UUID").
-			Where(sq.Eq{"pipelines.Status": model.PipelineReady})
+		query := opts.AddFilterToSelect(sqlBuilder).From("pipelines").
+			LeftJoin("pipeline_versions ON pipelines.DefaultVersionId = pipeline_versions.UUID")
+		if filterContext.ReferenceKey != nil && filterContext.ReferenceKey.Type == common.Namespace {
+			glog.Info("Using Namespace to filter the query")
+			query = query.Where(
+				sq.Eq{"pipelines.Status": model.PipelineReady,
+					"pipelines.Namespace": filterContext.ReferenceKey.ID},
+			)
+		} else {
+			query = query.Where(
+				sq.Eq{"pipelines.Status": model.PipelineReady},
+			)
+		}
```

Comment:
Is this if/else needed? From what I understand, the `filterContext` always contains a namespace reference key, because the `ListPipelines` endpoint sets it to the default value if it's not present.

Reply:
We still need to check this, otherwise you'll get a nill pointer while checking for the ID. Same on the List Pipelines method.